pandas/core/groupby/groupby.py

```diff
@@ -992,6 +983,17 @@ def _reset_group_selection(self) -> None:
             self._group_selection = None
             self._reset_cache("_selected_obj")
 
+    @contextmanager
+    def _group_selection_context(self) -> Iterator[GroupBy]:
```

Comment:
wouldnt' object to doing this change (e.g. to _group_selection_context) as a pre-cursor PR

Reply:
> 
> 
> wouldnt' object to doing this change (e.g. to _group_selection_context) as a pre-cursor PR

Done #43796